src/main/scala/Core/RegisterFile.scala

```diff
@@ -66,14 +66,16 @@ class RegisterFile(val threads: Int, val readPorts: Int = 2, val writePorts: Int
     // We also need to register the data and addresses since we are returning
     // last cycle's requests.
     val readUnderWrites = (writeIndexes zip io.write).map { case (writeIndex, writePort) =>
-      (RegNext(writeIndex) -> RegNext(writePort.data))
+      (RegNext(writeIndex) ->
```

Comment:
This looks like an unrelated change

Reply:
Right... These are the changes from #31, which the current branch is built on top of. Do you think we should combine these PRs into one PR, or keep them separate?